target/linux/airoha/Makefile

```diff
@@ -3,13 +3,11 @@ include $(TOPDIR)/rules.mk
 ARCH:=arm
 BOARD:=airoha
 BOARDNAME:=Airoha ARM
-CPU_TYPE:=cortex-a7
-FEATURES:=dt squashfs nand ramdisk gpio source-only
+SUBTARGETS:=en7523 en7581
```

Comment:
On some other vendors we follow a scheme where we try to group SoCs by CPU.

So in lack of a better suggestion, I suggest: 
* en7523 => cortex-a7
* en7581 => cortex-a53

The reason why I am suggesting this is that 
* en7523 subtarget is good for EN7523, EN7529 and EN7561
* en7581 subtarget is good for EN7551. EN7566 and EN7581

Reply:
It depends on how similar these SoCs are. I do not have much overview over the details of the airoha SoCs.

Reply:
Honesty group by CPU type doesn't seems so clear and might be confusing... also differnet soc probably needs different load values hence it's better to have different subtarget and .mk honestly.

Reply:
> differnet soc probably needs different load values hence it's better to have different subtarget and .mk honestly.

Not from my experience with the SoCs listed above. These SoCs are very, very similar. They are 95% identical. Except the functionality that is not present those SoCs are able to use the same device tree, U-Boot etc, because register space is the same except the added/removed IP blocks. For illustration: EN7529 is an improved EN7523, EN7562 is like EN7523/29 but without PoN (I think that one might support DSL). I can look up the details for EN7581 if needed as well.

I think having a different subtarget and compiling a different kernel (and nonshared packages) seems way overkill and is a waste of resources. We do not do it for Allwinner, Rockchip or NXP and the SoCs supported by those targets differ way more significantly from another than these Airoha SoCs do.

>  Honesty group by CPU type doesn't seems so clear and might be confusing... 

The world is complex/confusing. The fact that Airoha does not have public product pages for those variants does not make it easier. So confusing, that I made a mistake in my post above and confused EN7562 and EN7561... 😊

I do not have a a very strong opinion on the name of the subtarget itself, our opinion just differs on what is more confusing SoC-number or CPU type.  🤷‍♂️ 😊 Still, to clear up the confusion as much as possible, the human-friendly name should at least reveal that the one subtarget is for all three SoCs. I have opened https://github.com/openwrt/openwrt/pull/16785 for that. Let me know what you think.

Reply:
Well you need to also consider the other side, aka google and search engine.

Searching a SoC for the cpu name will never reference the SoC itself.
Also they are 95% identical but still not supported by OpenWrt as no DTS and nobody have actual device.
I agree on the waste of resource about the non-shared packages but I feel making a SoC hard to find online for the sake of futureproofing it (if and when other SoC support will be introduced) seems a bit too much.

Aside from this yes you PR is great and very much at least signal that those SoC are 95% similar.